lib/gs-plugin-loader.c

```diff
@@ -1916,16 +1890,10 @@ GsAppList *
 gs_plugin_loader_get_pending (GsPluginLoader *plugin_loader)
 {
 	GsPluginLoaderPrivate *priv = gs_plugin_loader_get_instance_private (plugin_loader);
-	GsAppList *array;
-	guint i;
+	g_autoptr(GMutexLocker) locker = g_mutex_locker_new (&priv->pending_apps_mutex);
```

Comment:
Again, why the change to g_mutex_locker_new rather than just keep the g_mutex_lock / g_mutex_unlock?  And, I suppose it doesn't really matter, but do you intentionally include the gs_app_list_new() within the mutex where it used to be outside?

Reply:
It's because it's better to use the RAII, and I modified most of the function so I thought it'd be nice to sneak in this change. I will change the order though as you pointed out.